src/core/thread/mle.cpp

```diff
@@ -2660,6 +2660,14 @@ void Mle::HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aMessageIn
 #endif
 #endif // OPENTHREAD_FTD
 
+#if OPENTHREAD_MTD
+#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
```

Comment:
Combine the `#if` checks?
```suggestion
#if OPENTHREAD_MTD && OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
```

Reply:
I did it at first because I thought there might be more cases in the future inside the OPENTHREAD_MTD but there's not much chance and your suggestion is more concise. Modified as requested. 
Thanks for the review.